source/Halibut/Transport/TcpConnectionFactory.cs

```diff
@@ -52,16 +54,22 @@ public async Task<IConnection> EstablishNewConnectionAsync(ExchangeProtocolBuild
             var certificateValidator = new ClientCertificateValidator(serviceEndpoint);
             var client = await CreateConnectedTcpClientAsync(serviceEndpoint, log, cancellationToken);
             log.Write(EventType.Diagnostic, $"Connection established to {client.Client.RemoteEndPoint} for {serviceEndpoint.BaseUri}");
-
-            var stream = client.GetStream();
+            
+            var networkStream = client.GetStream();
 
             log.Write(EventType.SecurityNegotiation, "Performing TLS handshake");
-            var ssl = new SslStream(stream, false, certificateValidator.Validate, UserCertificateSelectionCallback);
+
+#if NETFRAMEWORK
```

Comment:
Could `AuthenticateAsClientEnforcingTimeout` contain the `#if ; #else`  stuff as well so it's all wrapped up in one extension method?